Modified ActarsimDetectorConstriction in order to handle both ActarTPC and SpecMAT configurations

tmarchi opened the pull request:
A main switch was added (/ActarSim/det/AT_ProjectSwitch). Default value is Actar_TPC that allows for backward compatibile behavior.
ActarsimDetectorConstruction::Construct() was modified to define only once the world volume and switch between different projects.
A test was done using 12C_batch.mac and the results seem compatible with earlier version.

Commit:
Modified ActarsimDetectorConstriction in order to handle both ActarTPC and SpecMAT configurations
A main switch was added /ActarSim/det/AT_ProjectSwitch. Default value is Actar_TPC that allows for backward compatibile behavior.
ActarsimDetectorConstruction::Construct() was modified to define only once the world volume and switch between different projects.
A test was done using 12C_batch.mac and the results seem compatible with earlier version.

hapol commented:
Please, note the following comments and execute the required actions before merging the code under the merge request:

 1- (mandatory) files permission have been modified (100644 → 100755) which could cause security problems (execution permission). Default permissions should remain 644 for most of the files, except libraries, executables, or other special files.

2- (mandatory) detector-dependent switches are defined along the code by G4String with possible values on/off (as in G4String ACTARTPCDEMOGeoIncludedFlag; ). The code related with variable G4int AT_Project; [include/ActarSimDetectorConstruction.hh, line 117] breaks this symmetry. The suggested action is to use a G4String equivalent variable (as G4String AT_ProjectGeoIncludedFlag; ) and modify accordingly the code in include/ActarSimDetectorMessenger.hh and src/ActarSimDetectorMessenger.cc.
The modification is mandatory cause it breaks present functionality (selecting ACTARTPCDemo is not guaranteed by the command  /ActarSim/det/AT_ProjectSwitch  Actar_TPC ).

3- (optative)  A G4VPhysicalVolume\* ConstructSpecMAT(); function is included to have a separate Construct function for the project. This is, in principle, different to all other detectors (as Maiko, ACTARTPCDEMO, ... are all included in the generic ConstructActar(). Nevertheless, as it is an internal modification which does not modify any interface, it could be allowed to have a separate development. 
Alternatively you could implement in the general ConstructActar() the AT_ specific code under the flag AT_ProjectGeoIncludedFlag. In any case, please, note that ConstructActar() is not only for ACTAR TPC, but for any predefined chamber or for any geometry constructed on the fly. Other option (I would prefer) would be, according to your proposal, to create additional ConstructACTARTPCDEMO(), ConstructACTARTPC(), ConstructMAIKO(), ConstructOthers(), ... In this way the separation for each predefined geometry would be complete.

4- (mandatory) According to the previous comment (3), please do not modify the existing ConstructActar() for ConstructActarTPC(), as the function includes not "only" ActarTPC, but actually all possible geometries. This modification would be possible after a complete separation per detector (as commented before).

Commit:
Major modification to the DetectorConstruction functionality.
Now the Construct method acts as a main switch between the different main geometrical configurations: ACTAR_TPC, ACTARTPC DEMO, SpecMAT, Maiko, Others
This does not affect the user interface.

Commit:
splitting the different geometries in dedicated ConstructXXXXX functions

Commit:
fixing file permissions and adding modified macro

Commit:
again with file permissions and demo macro

tmarchi commented:
Splitting of the Construct() method was completed according to requests 2,3,4.
